test/Microsoft.TemplateEngine.Edge.UnitTests/TemplateCreatorTests.cs

```diff
@@ -670,12 +671,68 @@ public async void InstantiateAsync_ConditionalParametersIsRequiredEvaluation(boo
                 parameters2: parameters);
         }
 
+        private const string TemplateConfigPreferDefaultNameWithDefaultName = /*lang=json*/ """
+            {
+                "identity": "test.template",
+                "name": "tst",
+                "shortName": "tst",
+                "preferDefaultName": true,
+                "defaultName": "defaultName",
+                "sourceName": "sourceFile"
+            }
+            """;
+
+        private const string TemplateConfigPreferDefaultNameWithoutDefaultName = /*lang=json*/ """
+            {
+                "identity": "test.template",
+                "name": "tst",
+                "shortName": "tst",
+                "preferDefaultName": true,
+                "sourceName": "sourceFile"
+            }
+            """;
+
+        private const string TemplateConfigNoPreferDefaultNameWithDefaultName = /*lang=json*/ """
+            {
+                "identity": "test.template",
+                "name": "tst",
+                "shortName": "tst",
+                "preferDefaultName": false,
+                "sourceName": "sourceFile"
+            }
+            """;
+
+        [Theory]
+        [InlineData(TemplateConfigPreferDefaultNameWithDefaultName, "thisIsAName", "./thisIsAName.cs", false, "")]
+        [InlineData(TemplateConfigPreferDefaultNameWithDefaultName, null, "./defaultName.cs", false, "")]
+        [InlineData(TemplateConfigNoPreferDefaultNameWithDefaultName, null, "./tst2.cs", false, "")]
+        [InlineData(TemplateConfigPreferDefaultNameWithoutDefaultName, null, "./tst2.cs", true, "Failed to create template due naming issues")]
```

Comment:
Is this error message correct? Where it's coming from? Should it be the one from `LocalizableStrings` now?